R/app.R

```diff
@@ -0,0 +1,251 @@
+              "brain & ons" = "brain",
+              "colon & rectum" = "colon",
+              "esophagus",
+              "kidney & renal pelvis" = "kidney",
+              "leukemia",
+              "liver & bile duct" = "liver",
+              "lung & bronchus" = "lung",
+              "melanoma of the skin" = "melanoma",
+              "non-hodgkin lymphoma" = "lymphoma",
+              "oral cavity & pharynx" = "oral",
+              "pancreas",
+              "stomach",
+              "thyroid"
+            ),
+            selected = "allsites"
+          ),
+
+          selectInput(
+            "race",
+            "Select Race:",
+            choices = c(
+              "All Races (includes Hispanic)" = "allraces",
+              "White (non-Hispanic)" = "white",
+              "Black (non-Hispanic)" = "black",
+              "American Indian / Alaska Native (non-Hispanic)" = "native",
+              "Asian / Pacific Islander (non-Hispanic)" = "asian",
+              "Hispanic (Any Race)" = "hisp"
+            ),
+            selected = "allraces"
+          ),
+
+          selectInput(
+            "sex",
+            "Select sex:",
+            choices = c(
+              "both sexes" = "both",
+              "males",
+              "females"
+            ),
+            selected = "both"
+          ),
+
+          selectInput(
+            "age",
+            "Select age range:",
+            choices = c(
+              "all ages" = "all",
+              "ages <50" = "<50",
+              "ages 50+" = "50+",
+              "ages <65" = "<65",
+              "ages 65+" = "65+",
+              "ages <15" = "15",
+              "ages <20" = "20"
+            ),
+            selected = "all"
+          ),
+
+          selectInput(
+            "stage",
+            "Select cancer stage:",
+            choices = c(
+              "all stages" = "allstages",
+              "late stage (regional & distant)" = "latestage"
+            ),
+            selected = "allstages"
+          ),
+
+          selectInput(
+            "year",
+            "Select time span:",
+            choices = c(
+              "latest 5 year average" = "5yr",
+              "latest single year (us by state)" = "1yr"
+            ),
+            selected = "5yr"
+          )
+        ),
+
+        # Show the generated choropleth plot
+        mainPanel(
+           leafletOutput("choropleth")
+        )
+    )
+)
+
+# Define server logic
+server <- function(input, output) {
+
+    # Generate a choropleth plot using {leaflet}
+    output$choropleth <- renderLeaflet({
+      chosen_cancer = input$cancer_type
+      race = input$race
+      sex = input$sex
+      age = input$age
+      stage = input$stage
+      year = input$year
+
+      # Create a new DuckDB corresponding to incidence data from US states
+      con <- dbConnect(duckdb::duckdb(), "cancer-incidence-usa-state.duckdb")
+
+      ## 1. Ingest data from State Cancer Profiles if needed
```

Comment:
Data ingestion and cleaning should take place outside of the shiny app itself, the app logic should assume that the data is completely prepared for the app. This make it easier to iterate on data cleaning without necessarily touching the app.